booster/library/Booster/Pattern/Rewrite.hs

```diff
@@ -52,14 +64,20 @@ import Booster.Pattern.ApplyEquations qualified as ApplyEquations
 import Booster.Pattern.Base
 import Booster.Pattern.Bool
 import Booster.Pattern.Index (TermIndex (..), kCellTermIndex)
-import Booster.Pattern.Match
+import Booster.Pattern.Match (
+    FailReason (ArgLengthsDiffer, SubsortingError),
+    MatchResult (MatchFailed, MatchIndeterminate, MatchSuccess),
+    MatchType (Rewrite),
+    SortError,
+    Substitution,
+    matchTerms,
+ )
 import Booster.Pattern.Util
 import Booster.Prettyprinter
 import Booster.SMT.Interface qualified as SMT
 import Booster.Util (Flag (..), constructorName)
-import Data.Coerce (coerce)
 
-newtype RewriteT io err a = RewriteT
+newtype RewriteT err io a = RewriteT
```

Comment:
(off-topic) I think we can inline `err ~ RewriteFailed "Rewrite"` here (and maybe drop the `"Rewrite"`, too...)